### PR TITLE
Actions — Part 2

### PR DESCRIPTION
**Pull request by radex**
Adds a database constructor flag `actionsEnabled: boolean`. If `true`, you can only perform write operations (create/update/delete) from inside an action. Plus tests.

Ultimately, this will always be enabled, but this way we can experiment and iterate without forcing changes onto users before we're sure this is the right way to go

plus some tests refactoring

### CODE REVIEW

**Comment by radex** [comment]
@kokusGr CR

File: src/Database/index.js [hunk 1]
```diff
@@ -88,4 +93,8 @@ export default class Database {
       collection.unsafeClearCache()
     })
   }
+
+  _ensureInAction(error: string): void {
```
Comment (kokusGr) [inline src/Database/index.js]: Hmm is this safe? I mean there might be other actions in queue and I could use `Collection.create` outside of an action and this wouldn't throw an error. Am I right? :P Sometimes it's kinda hard to follow the flow in async code :/

File: src/Model/test.js [hunk 2]
```diff
@@ -279,6 +263,96 @@ describe('watermelondb/Model', () => {
...
+    await expectToRejectWithMessage(
+      model.markAsDeleted(),
+      /can only be called from inside of an Action/,
+    )
+
+    // check that no throw inside action
+    await database.action(async () => {
+      await model.update(noop)
+      await model.markAsDeleted()
+      await model.markAsDeleted()
```
Comment (kokusGr) [inline src/Model/test.js]: I think one of them should be `destroyPermanently` :)

Comment (radex) [reply src/Model/test.js]: nice catch!

File: src/Database/index.js (hunk 1 above)
Comment (radex) [reply src/Database/index.js]: You're right! Asynchronicity is really hard! This solution does not give you 100% guarantee of correctness, but you should be able to catch 99% errors, because you'd need to constantly have a long-running action to not hit the error.

I think the only way to guarantee correctness would be to remove batch/create/update methods, and have those methods only on an object passed to the `database.action()` block

Comment (kokusGr) [reply src/Database/index.js]: Ok, thanks for explanation :) I guess 99% is good enough for now :)
